odoo/addons/base/models/ir_translation.py

```diff
@@ -110,8 +110,8 @@ def finish(self):
                            AND noupdate IS NOT TRUE
                            ON CONFLICT (type, lang, md5(src)) WHERE type = 'code'
                             DO UPDATE SET (name, lang, res_id, src, type, value, module, state, comments) = (EXCLUDED.name, EXCLUDED.lang, EXCLUDED.res_id, EXCLUDED.src, EXCLUDED.type, EXCLUDED.value, EXCLUDED.module, EXCLUDED.state, EXCLUDED.comments)
-                            WHERE EXCLUDED.value IS NOT NULL AND EXCLUDED.value != '';
-                       """ % (self._model_table, self._table))
+                            WHERE %s.comments != 'openerp-web' AND EXCLUDED.value IS NOT NULL AND EXCLUDED.value != '';
```

Comment:
Won't this break updates of translations when the translation entry already exists for a JS term (e.g. because it was previously generated manually or via the `base.language.export` wizard), even though the translation value may be empty or wrong?

Reply:
Indeed,
if we have an untranslated js-term in database, then it'd not be overrided

What about this condition?

```suggestion
                            WHERE (EXCLUDED.comments = 'openerp-web' OR %s.comments != 'openerp-web' OR %s.value IS NULL OR %s.value='') AND EXCLUDED.value IS NOT NULL AND EXCLUDED.value != '';
```

Reply:
I applied my suggestion in this PR

Reply:
With that updated condition I still see chances for skipping valid translation updates, and chances of losing the "openerp-web" comment ;-)

Now if we take a step back, the problem is that we're using that `comments` field as a flag for "load on frontend". And since we're merging translations and overwriting comments, we sometimes lose the flag.

By the way, once loaded into the database, we don't care about PO comments, *except* for that special "openerp-web" flag. This is true regardless of the possible legacy comments you can find in older databases (I've seen one with 800+ different values for `comments`).

Based on that, a more robust fix might be to detect the flag specifically and preserve it, while leaving the rest of the logic for updating translations intact.

What I mean is something like this (untested pseudo patch), basically replacing the blind overwrite of `comments` with one that preserves the flag:


```patch
--- odoo/addons/base/models/ir_translation.py
+++ odoo/addons/base/models/ir_translation.py
@@ -119,9 +119,10 @@ class IrTranslationImport(object):
                            WHERE type = 'model'
                            AND noupdate IS NOT TRUE
                            ON CONFLICT (type, lang, name, res_id) WHERE type = 'model'
-                            DO UPDATE SET (name, lang, res_id, src, type, value, module, state, comments) = (EXCLUDED.name, EXCLUDED.lang, EXCLUDED.res_id, EXCLUDED.src, EXCLUDED.type, EXCLUDED.value, EXCLUDED.module, EXCLUDED.state, EXCLU>
+                            DO UPDATE SET (name, lang, res_id, src, type, value, module, state, comments) = (EXCLUDED.name, EXCLUDED.lang, EXCLUDED.res_id, EXCLUDED.src, EXCLUDED.type, EXCLUDED.value, EXCLUDED.module, EXCLUDED.state,
+                                                                                                             CASE WHEN %s.comments = 'openerp-web' THEN 'openerp-web' ELSE EXCLUDED.comments END)
                             WHERE EXCLUDED.value IS NOT NULL AND EXCLUDED.value != '';
-                       """ % (self._model_table, self._table))
+                       """ % (self._model_table, self._table, self._table))
             count += cr.rowcount
 
             cr.execute(""" INSERT INTO %s(name, lang, res_id, src, type, value, module, state, comments)
```

What do you think?

Reply:
CASE WHEN %s.comments = 'openerp-web' THEN 'openerp-web' ELSE EXCLUDED.comments END

So, this means "apply updates as before, but keep flag openerp-web if it's set".

Indeed, It looks more robust than

    WHERE (EXCLUDED.comments = 'openerp-web' OR %s.comments != 'openerp-web' 
                    OR %s.value IS NULL OR %s.value='') AND ...

The only problem with the proposed solution I see is for cases when same term has different translations for js a py sides. After the patch, the py  translation may be used as js translation.

@mart-e what do you think?

Reply:
I applied updates from #60765 and make some research about english terms that have different translation in fronted backend.

```
devel=# select src, value, comments, name, lang, type from ir_translation where src='Free' AND lang='fr_BE' AND type='code';
 src  |  value   |  comments   |                                name                                 | lang  | type 
------+----------+-------------+---------------------------------------------------------------------+-------+------
 Free | Gratuit  | openerp-web | addons/web_editor/static/src/js/wysiwyg/widgets/crop_dialog.js      | fr_BE | code
 Free | Original |             | addons/event_sale/models/event.py                                   | fr_BE | code
 Free | Gratuit  | openerp-web | addons/website_sale_delivery/static/src/js/website_sale_delivery.js | fr_BE | code

devel=# Select Src, Value, Comments, Name, Lang, Type From Ir_Translation Where Src='Free' AND Lang='es_ES' AND Type='code';
 src  |  value   |  comments   |                                name                                 | lang  | type 
------+----------+-------------+---------------------------------------------------------------------+-------+------
 Free | Libre    | openerp-web | addons/web_editor/static/src/js/wysiwyg/widgets/crop_dialog.js      | es_ES | code
 Free | Libre    |             | addons/event_sale/models/event.py                                   | es_ES | code
 Free | Gratuito | openerp-web | addons/website_sale_delivery/static/src/js/website_sale_delivery.js | es_ES | code


devel=#  select src, value, comments, name, lang, type from ir_translation where src='Applications' AND lang='fr_BE' AND type='code';
     src      |    value     |  comments   |                      name                      | lang  | type 
--------------+--------------+-------------+------------------------------------------------+-------+------
 Applications | Candidatures |             | addons/hr_recruitment/models/hr_recruitment.py | fr_BE | code
 Applications | Applications | openerp-web | addons/web_enterprise/static/src/xml/base.xml  | fr_BE | code
(2 rows)


```

So, there is just a few problematic terms.  And in many languages the translators didn't make distinguishes, so I don't see a big problem if one term is used instead of another. But I'd make a priority to frontend translation, because of term "Free": we don't want to see "Original (fr) /Libre (es)" where it should be "Gratuit (fr)/ Gratuito (es)" (module website_sale_delivery). If we agree on that, we should stay on my approach, rather than marking backend translation as frontend translation.

Reply:
Alternative approach, mark duplicated terms as openerp-web, e.g. use 

    CASE WHEN %s.comments = 'openerp-web' THEN 'openerp-web' ELSE EXCLUDED.comments END

but, make different english terms for word "Free", which means to break some part of translations in stable